Stack/Opc.Ua.Core/Types/Utils/HiResClock.cs

```diff
@@ -20,47 +20,40 @@ namespace Opc.Ua
     /// <summary>
     /// Produces high resolution timestamps.
     /// </summary>
-    public class HiResClock
+    public static class HiResClock
     {
-        /// <summary>
-        /// Returns the current UTC time (bugs in HALs on some computers can result in time jumping backwards).
-        /// </summary>
-        public static DateTime UtcNow
+        // In Net Core 3.0+ has monotonic timer from system start.
+        // https://docs.microsoft.com/ru-ru/dotnet/api/system.environment.tickcount64?view=netcore-3.1
+#if NETCORE3_0 || NETCORE3_1
+        public static long TickCount64 { get { return Environment.TickCount64; } } 
+#else
+
+        /// <summary>Get monotonic timer from system starts in Windows in Msec. In other systems using DateTime. (posible lost monitored items when combuter time shift. see #1121)</summary>
+        public static long TickCount64
         {
             get
             {
-                return DateTime.UtcNow;                
+                if (Environment.OSVersion.Platform == PlatformID.Win32NT)
+                    return GetTickCount64();
+                else
+                    return DateTime.UtcNow.Ticks / TimeSpan.TicksPerMillisecond;
             }
         }
 
-        /// <summary>
-        /// Disables the hi-res clock (may be necessary on some machines with bugs in the HAL).
-        /// </summary>
-        public static bool Disabled
-        {
-            get
-            {
-                return s_Default.m_disabled;
-            }
+        [System.Runtime.InteropServices.DllImport("kernel32.dll")]
```

Comment:
I hope there are no side effects on Linux, I'd just add a #If NETFRAMEWORK here to make sure there are no issues on other platforms.